src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs

```diff
@@ -56,31 +56,29 @@ internal class LaunchProfilesDebugLaunchProvider : DebugLaunchProviderBase, IDep
         /// </summary>
         public async Task<bool> CanBeStartupProjectAsync(DebugLaunchOptions launchOptions)
         {
-            ILaunchProfile activeProfile = await GetActiveProfileAsync();
+            if (await GetActiveProfileAsync() is ILaunchProfile activeProfile)
```

Comment:
This looks like a runtime type check, which it isn't. What do you think of using the non-null pattern here instead?

```suggestion
            if (await GetActiveProfileAsync() is {} activeProfile)
```

It's a shame the `not null` pattern doesn't work with declarations like this.

Reply:
That's not my favorite syntax, honestly--it obscures the type information in the same way as `var`. If it was already the predominant pattern in the repo I would stick with it, though.

Checking the generated IL, the `is ILaunchProfile activeProfile` doesn't incur a runtime type check--it's actually the same IL as `is {} activeProfile`.

Reply:
My concern with this syntax is that it looks like a runtime type check, which it isn't.

Wanting the type to be explicit is fair, but when I read this code I worry that `GetActiveProfileAsync` returns some other type, not shown in the code, and then the value is subjected to a type test.

I'd keep it simple and use the following which is unambiguous, and is by far the dominant pattern in the repo:

```suggestion
            ILaunchProfile activeProfile = await GetActiveProfileAsync();
            if (activeProfile is not null)
```